File: incidents/src/main/java/fr/openent/incidents/service/SeriousnessService.java
package fr.openent.incidents.service;

import fr.wseduc.webutils.Either;
import io.vertx.core.Future;
import io.vertx.core.Handler;
import io.vertx.core.json.JsonArray;
import io.vertx.core.json.JsonObject;

public interface SeriousnessService {
    /**
     * get seriousnesses
     *
     * @param structureId   Structure Identifier
     * @param handler       Function handler returning data
     */
    void get(String structureId, Handler<Either<String, JsonArray>> handler);

    /**
     * create seriousness
     *
     * @param seriousnessBody       seriousnessBody fetched
     */
    Future<JsonObject> create(JsonObject seriousnessBody);

    /**
     * update seriousness
     *
     * @param seriousnessBody   seriousnessBody fetched
     */
    Future<JsonObject> put(JsonObject seriousnessBody);

    /**
     * delete seriousness
     *
     * @param seriousnessId seriousness Identifier
     * @param handler       Function handler returning data
     */
    void delete(Integer seriousnessId, Handler<Either<String, JsonObject>> handler);
}

File: incidents/src/main/resources/public/ts/services/SeriousnessService.ts
import {ng} from 'entcore'
import { http, HttpResponse } from 'entcore-toolkit';

export interface Seriousness {
    id: number;
    structure_id?: string;
    label: string;
    level: number;
    hidden: boolean;
    used?: boolean;
    exclude_alert_seriousness: boolean;
}

export interface SeriousnessRequest {
    id?: number;
    structureId?: string,
    label?: string;
    level?: number;
    hidden?: boolean;
    excludeAlertSeriousness?: boolean;
}

export interface SeriousnessService {
    get(structureId: string): Promise<Seriousness[]>;
    create(seriousnessBody: SeriousnessRequest): Promise<HttpResponse>;
    update(seriousnessBody: SeriousnessRequest): Promise<HttpResponse>;
    delete(seriousnessId: number): Promise<HttpResponse>;
}

export const seriousnessService : SeriousnessService = {
    get: async (structureId: string): Promise<Seriousness[]> => {
        try {
            const {data} = await http.get(`/incidents/seriousnesses?structureId=${structureId}`);
            return data;
        } catch (err) {
            throw err;
        }
    },

    create: async (seriousnessBody: SeriousnessRequest): Promise<HttpResponse> => {
        return await http.post(`/incidents/seriousness`, seriousnessBody);
    },

    update: async (seriousnessBody: SeriousnessRequest): Promise<HttpResponse> => {
        return await http.put(`/incidents/seriousness`, seriousnessBody);
    },

    delete: async (seriousnessId: number): Promise<HttpResponse> => {
        return await http.delete(`/incidents/seriousness?id=${seriousnessId}`);
    },
};

export const SeriousnessService = ng.service('SeriousnessService', (): SeriousnessService => seriousnessService);
